config/cli.go

```diff
@@ -244,3 +254,87 @@ func validateHost(host string) error {
 
 	return nil
 }
+
+func readTOMLConfig(cfg *CLIConfig) error {
+	tomlCfgFile := "config.toml"
+	if _, err := os.Stat(tomlCfgFile); err == nil {
+		content, err := os.ReadFile(tomlCfgFile)
+		if err != nil {
+			return fmt.Errorf("failed to read TOML config: %w", err)
+		}
+
+		var tomlCfg TOMLConfig
+		if err := toml.Unmarshal(content, &tomlCfg); err != nil {
+			return fmt.Errorf("error parsing TOML: %v", err)
+		}
+
+		cfg.AdminPort = tomlCfg.AdminPort
+		cfg.L1Port = tomlCfg.L1Port
+		cfg.L2StartingPort = tomlCfg.L2StartingPort
+		cfg.InteropAutoRelay = tomlCfg.InteropAutoRelay
+		cfg.InteropDelay = tomlCfg.InteropDelay
+		cfg.LogsDirectory = tomlCfg.LogsDirectory
+		cfg.L1Host = tomlCfg.L1Host
+		cfg.L2Host = tomlCfg.L2Host
+
+		if tomlCfg.ForkConfig != nil {
+			cfg.ForkConfig = &ForkCLIConfig{}
+			if tomlCfg.ForkConfig.L1ForkHeight != 0 {
+				cfg.ForkConfig.L1ForkHeight = tomlCfg.ForkConfig.L1ForkHeight
+			}
+			if len(tomlCfg.ForkConfig.Chains) > 0 {
+				cfg.ForkConfig.Chains = tomlCfg.ForkConfig.Chains
+			}
+			if tomlCfg.ForkConfig.Network != "" {
+				cfg.ForkConfig.Network = tomlCfg.ForkConfig.Network
+			}
+			if tomlCfg.ForkConfig.InteropEnabled {
+				cfg.ForkConfig.InteropEnabled = tomlCfg.ForkConfig.InteropEnabled
+			}
+
+			if cfg.ForkConfig.L1ForkHeight == 0 && len(cfg.ForkConfig.Chains) == 0 && cfg.ForkConfig.Network == "" {
+				cfg.ForkConfig = nil
+			}
+		}
+	} else if !os.IsNotExist(err) {
+		return fmt.Errorf("error checking TOML config file: %w", err)
+	}
+
+	return nil
+}
+
+func populateFromCLIContext(cfg *CLIConfig, ctx *cli.Context) {
```

Comment:
RC: this function makes it so that if the TOML config is set, the cli flags can't override it. Should update it so that the explicit flags passed to CLI always take precedence. That would align better with the expected behavior of most CLIs (explicit flags override file base config)

We can just remove the checks for the default values (ie. cfg.L1Port == 0)